bee-ternary/src/bigint/overflowing_add.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright 2020 IOTA Stiftung
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and limitations under the License.
+
+//! Extensions to `overflowing_add`.
+
+pub(crate) trait OverflowingAdd<T = Self> {
+    type Output;
```

Comment:
I wonder why we need this associated type. For example the API of `u32` is:
`pub const fn overflowing_add(self, rhs: u32) -> (u32, bool)`
As you can further down below in the implementation of the trait method, `sum` has the same type as `rhs`, but the first element of the returned tuple is `Self::Output`. Where would it make sense to have a different type here?

Reply:
I'm sure Janis had a good reason but idk. Let's remove.